### PR TITLE
Centralise task filtering into own component; list task states in logical order

### LINKED ISSUE #1118: view toolbar: task state filter should list task states in logical order
The task state filter used in some of the views (e.g. tree view) lists task states in alphabetical order.

This isn't easy to read, suggest using logical state order intended for display purposes which is defined here:

https://github.com/cylc/cylc-ui/blob/35a39e3c55912cecddd07be94067a9e24d092568/src/model/TaskState.model.js#L47-L56

**Pull requests welcome!**

### PR DESCRIPTION
**Pull request by MetRonnie**
- The task filtering HTML and code was duplicated in the Tree and Table components. I have extracted it into its own component and simplified the code
- Closes #1118

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate tests are included (unit and/or functional).
- [x] No change log entry required (mostly invisible to users).
- [x] No documentation update required.

### CODE REVIEW

**Comment by oliver-sanders** [comment]
Due to the refactoring this will be a tricky one to reconcile with the graph view / data store work as filtering works in a different way when views are powered from the central data store.

**Comment by hjoliver** [comment]
> Due to the refactoring this will be a tricky one to reconcile with the graph view / data store work ...

What's your suggestion - shelve this one?

**Comment by MetRonnie** [comment]
I have looked at #1108 and I don't see too much surface area for conflict, only a handful of lines

**Comment by oliver-sanders** [comment]
I haven't got filtering working in all views on that branch yet. The main difference is that in order to drive the views off of the central data store we cannot change the data or set attributes on it directly. Instead we must store the filter state in the view locally.

I'm going to try to get back to that branch this week to see if I can clear out the remaining issues and find out.

**Comment by MetRonnie** [comment]
Note that currently this new filter component does not house the logic of the filtering, that is left up to the parent components that use/embed the filter component

**Comment by MetRonnie** [comment]
Converting to draft, will be easier to deconflict this one after #1108 has gone in

File: src/components/cylc/common/filter.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/**
```
Comment (oliver-sanders) [inline src/components/cylc/common/filter.js:1]: Perhaps move into src/utils?

**Review by oliver-sanders — APPROVED** [review]
LGTM

**Review by wxtim — APPROVED** [review]
- [x] Read code.
- [x] Tried functionality.

**Pull request merged**